### ISSUE TITLE
Merge Kotlin and Kotlin multiplaform into one

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-docs#11848]
### SDK

Kotlin Multiplatform SDK

### Description

We have two entries on the docs for Kotlin:
![Image](https://github.com/user-attachments/assets/07cdaf5c-5142-48d7-986c-be2950e6464c)

It takes space in the landing page as well as the drop down. 

Kotlin docs is only a landing page to route the reader to either Android, Java or Kotlin Multi-platform: https://docs.sentry.io/platforms/kotlin/
![Image](https://github.com/user-attachments/assets/e629b71b-3cf0-4354-9eac-9aea0e5e0f5b)

It does that while showing `sentry-java` information the right side.
![Image](https://github.com/user-attachments/assets/170500e4-4610-4df2-bcc9-ddc20efb1692)

Kotlin extensions, which lives under the Java docs, apply to Kotlin in general, including KMP.
https://docs.sentry.io/platforms/java/enriching-events/scopes/#kotlin-coroutines

Kotlin itself has one doc: https://kotlinlang.org/
It mentions its capabilities, including KMP.

### Suggested Solution

I propose we merge both things into one area: Kotlin
Redirect all `kotlin-multiplatform` docs to `kotlin`, and move all pages there.

The landing page can have the same clarification that we have specific Android and Java (JVM) docs, including Spring Boot, elsewhere.

A smaller banner could live on top of all pages that are specific to KMP, with this clarification about Android and JVM docs.

@kahest @HazAT @buenaflor wdyt?

**Comment by kahest** [comment]
@bruno-garcia do you think having two items is confusing, or is the main point that it takes too much space? FWIW, KMP has much more traffic (2x that of Kotlin on the landing page), so I kinda like having KMP called out explicitly. No strong opinions against merging, but I wouldn't want to use sub-platforms/guides as we do for Apple, it's a bit annoying to maintain and only really usable if most of the content is the same IMO.

**Comment by bruno-garcia** [comment]
Yeah it was mainly odd to have 2 entries for what from the outside seems could be the same entry point.

Regarding the traffic I'd argue it's very likely the same traffic would exist under `platforms/kotlin/multiplatform` since the content would still exist, just the URL would change. We can redirect the existing URLs to the new path under `/kotlin` so links are not broken.

The part that makes it harder to justify changing things is: 

>  I wouldn't want to use sub-platforms/guides as we do for Apple, it's a bit annoying to maintain and only really usable if most of the content is the same IMO.

If there's no easy way to structure things, then I don't know how we would do this.

Maybe @chargome or @coolguyzone have ideas?

**Comment by coolguyzone** [comment]
Hey @bruno-garcia, I like this idea, I think the number of SDKs we list is a bit overwhelming, anything we can do to consolidate it would be great. It seems reasonable to assume that people looking for Kotlin Multiplatform will click on Kotlin, it might cost them one extra click to get the content they want, but that seems reasonable. @lizokm may have thoughts about this as well.

**Comment by kahest** [comment]
let's do it :)

**Comment by buenaflor** [comment]
no objections here 👍

**Comment by chargome** [comment]
Just want to mention that if you want Kotlin Multiplatform to appear in dropdowns etc. you will either need to keep it as a platform (like now) or as a sub-guide of Kotlin. Adding it as a guide will also open up the possibility of using a shared common folder, platform-includes etc. Otherwise it would be just a sub-page that has it's own content.
